components/external_server/src/model/actions.py

```diff
@@ -63,12 +65,14 @@ def move_item(
     container: Report | Subject | Metric, item_to_move: Subject | Metric | Source, new_position: Position
 ) -> tuple[int, int]:
     """Change the item position."""
+    ItemsDictType = MutableMapping[ItemId, Metric | Source | Subject]
```

Comment:
This method feels like it's trying to be very generic, while it's really just a simple mixin. Is there not a different way to do it?

Reply:
Yeah, I agree, but I didn't want to refactor, just get the typing correct. This is already becoming a huge change set and at least typing changes don't affect run time behavior.